govcd/api_vcd_test.go

```diff
@@ -545,6 +545,28 @@ func (vcd *TestVCD) removeLeftoverEntities(entity CleanupEntity) {
 			vcd.infoCleanup(notDeletedMsg, entity.EntityType, entity.Name, err)
 		}
 		return
+	case "user":
```

Comment:
I was also thinking how to make it smaller, but not sure if it's with this PR or do we make something separately to cleanup (if we ever will). A lot of it could be shrunken down based on the lookups.

One idea I had for load balancer components would be to rewrite it similarly:
```go
case "lbServiceMonitor", "lbServerPool", "other_component_having_same_requirements":  // share the object lookup part
 		if entity.Parent == "" {
			vcd.infoCleanup("removeLeftoverEntries: [ERROR] No parent specified '%s'\n", entity.Name)
			return
		}

    // these objects should be defined at function scope, rather than `switch` scope
		orgName, vdcName, edgeName := splitParent(entity.Parent, "|")

		org, vdc, edge, err = getOrgVdcEdgeByNames(vcd, orgName, vdcName, edgeName)
		if err != nil {
			vcd.infoCleanup("removeLeftoverEntries: [ERROR] %s \n", err)
		}
    fallthrough

 
case "lbServerPool":
		err = edge.DeleteLBServerPoolByName(entity.Name)
		if err != nil {
			vcd.infoCleanup(notFoundMsg, entity.EntityType, entity.Name)
			return
		}

		vcd.infoCleanup(removedMsg, entity.EntityType, entity.Name, entity.CreatedBy)
		return

case "lbServiceMonitor":
		err = edge.DeleteLBServiceMonitorByName(entity.Name)
		if err != nil {
			vcd.infoCleanup(notFoundMsg, entity.EntityType, entity.Name)
			return
		}

		vcd.infoCleanup(removedMsg, entity.EntityType, entity.Name, entity.CreatedBy)
		return
case "other_component_having_same_requirements":
		err = edge.DeleteLBOtherComponentByName(entity.Name)
		if err != nil {
			vcd.infoCleanup(notFoundMsg, entity.EntityType, entity.Name)
			return
		}

		vcd.infoCleanup(removedMsg, entity.EntityType, entity.Name, entity.CreatedBy)
		return

....
```

Reply:
Let's move it separate issue?

Reply:
I think this change requires its own PR

Reply:
Agreed on my side as well.